src/App.jsx

```diff
@@ -14,33 +16,94 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_ABC = 'abc';
+const SORT_BY_LENGTH = 'leng';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getGoodsView(sortField, reverse) {
+  let prepearedGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  switch (sortField) {
+    case SORT_ABC:
+      prepearedGoods = prepearedGoods.sort((cur, prew) => {
+        return reverse ? prew.localeCompare(cur) : cur.localeCompare(prew);
+      });
+      break;
+    case SORT_BY_LENGTH:
+      prepearedGoods = reverse
+        ? prepearedGoods.sort((a, b) => a.length - b.length).reverse()
+        : prepearedGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      prepearedGoods = reverse
+        ? prepearedGoods.reverse()
+        : [...goodsFromServer];
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+
+  const visibleGoods = getGoodsView(sortField, reverse);
+
+  const sortGoods = (sorting, reversing = false) => {
+    setReverse(reversing);
+    setSortField(sorting);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_ABC,
+          })}
+          onClick={() => sortGoods(SORT_ABC, reverse)}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SORT_BY_LENGTH,
+          })}
+          onClick={() => sortGoods(SORT_BY_LENGTH, reverse)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !reverse,
+          })}
+          onClick={() => sortGoods(sortField, !reverse)}
```

Comment:
The 'Reverse' button currently toggles the reverse state but does not affect the sorting field. Ensure that the reverse logic is correctly applied to the current sorting field.